### ISSUE TITLE
Blank page when accessing to http://localhost:8888/ (OSX 10.8.4)

### ISSUE THREAD

**Issue by RodrigoAyala** [SamyPesse/tv.js#27]
Hello guys,

I tried to run the project on a Mac (OSX 10.8.4), following the steps of the README, but when I go to http://localhost:8888/, I only see a blank page.

Also, the browser says that paths static/application.css and static/application.js doesn't exists (404 error).

Debug info:

**HTML of http://localhost:8888/:**:

```
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <title>Main</title>
        <link rel="stylesheet" href="static/application.css">
        <script type="text/javascript" src="static/application.js"></script>
    </head>
    <body></body>
</html>
```

```
$ make install
```

**Output of the previous command:**

```
/usr/local/bin/npm install .
npm http GET https://registry.npmjs.org/q/0.9.6
npm http GET https://registry.npmjs.org/underscore/1.4.4
npm http GET https://registry.npmjs.org/request/2.22.0
npm http GET https://registry.npmjs.org/express/3.0.0
npm http GET https://registry.npmjs.org/socket.io/0.9.16
npm http GET https://registry.npmjs.org/fluent-ffmpeg/1.5.2
npm http GET https://registry.npmjs.org/readable-stream/1.0.2
npm http GET https://registry.npmjs.org/peerflix
npm http 304 https://registry.npmjs.org/express/3.0.0
npm http 304 https://registry.npmjs.org/q/0.9.6
npm http 304 https://registry.npmjs.org/request/2.22.0
npm http 304 https://registry.npmjs.org/underscore/1.4.4
npm http 304 https://registry.npmjs.org/socket.io/0.9.16
npm http 304 https://registry.npmjs.org/fluent-ffmpeg/1.5.2
npm http 304 https://registry.npmjs.org/readable-stream/1.0.2
npm http 304 https://registry.npmjs.org/peerflix
npm http GET https://registry.npmjs.org/bitfield
npm http GET https://registry.npmjs.org/clivas
npm http GET https://registry.npmjs.org/range-parser/0.0.4
npm http GET https://registry.npmjs.org/hat/0.0.3
npm http GET https://registry.npmjs.org/mime
npm http GET https://registry.npmjs.org/numeral
npm http GET https://registry.npmjs.org/async
npm http GET https://registry.npmjs.org/read-torrent
npm http GET https://registry.npmjs.org/network-address/0.0.4
npm http GET https://registry.npmjs.org/peer-wire-protocol
npm http GET https://registry.npmjs.org/peer-wire-swarm
npm http GET https://registry.npmjs.org/part-file
npm http GET https://registry.npmjs.org/speedometer
npm http GET https://registry.npmjs.org/optimist
npm http GET https://registry.npmjs.org/once
npm http GET https://registry.npmjs.org/connect/2.6.0
npm http GET https://registry.npmjs.org/commander/0.6.1
npm http GET https://registry.npmjs.org/mkdirp/0.3.3
npm http GET https://registry.npmjs.org/cookie/0.0.4
npm http GET https://registry.npmjs.org/crc/0.2.0
npm http GET https://registry.npmjs.org/fresh/0.1.0
npm http GET https://registry.npmjs.org/methods/0.0.1
npm http GET https://registry.npmjs.org/send/0.1.0
npm http GET https://registry.npmjs.org/debug
npm http GET https://registry.npmjs.org/policyfile/0.0.4
npm http GET https://registry.npmjs.org/base64id/0.1.0
npm http GET https://registry.npmjs.org/socket.io-client/0.9.16
npm http GET https://registry.npmjs.org/redis/0.7.3
npm http GET https://registry.npmjs.org/qs
npm http GET https://registry.npmjs.org/json-stringify-safe
npm http GET https://registry.npmjs.org/forever-agent
npm http GET https://registry.npmjs.org/tunnel-agent
npm http GET https://registry.npmjs.org/http-signature
npm http GET https://registry.npmjs.org/aws-sign
npm http GET https://registry.npmjs.org/hawk
npm http GET https://registry.npmjs.org/oauth-sign
npm http GET https://registry.npmjs.org/cookie-jar
npm http GET https://registry.npmjs.org/node-uuid
npm http GET https://registry.npmjs.org/form-data/0.0.8
npm http GET https://registry.npmjs.org/requirejs/2.1.6
npm http GET https://registry.npmjs.org/commander/1.1.1
npm http GET https://registry.npmjs.org/wrench/1.5.1
npm http GET https://registry.npmjs.org/cli-color/0.2.2
npm http GET https://registry.npmjs.org/less/1.3.3
npm http GET https://registry.npmjs.org/watchr/2.4.3
npm http 304 https://registry.npmjs.org/bitfield
npm http 304 https://registry.npmjs.org/hat/0.0.3
npm http 304 https://registry.npmjs.org/mime
npm http 304 https://registry.npmjs.org/clivas
npm http 304 https://registry.npmjs.org/range-parser/0.0.4
npm http 304 https://registry.npmjs.org/numeral
npm http 304 https://registry.npmjs.org/async
npm http 304 https://registry.npmjs.org/read-torrent
npm http 304 https://registry.npmjs.org/peer-wire-protocol
npm http 304 https://registry.npmjs.org/network-address/0.0.4
npm http 304 https://registry.npmjs.org/peer-wire-swarm
npm http 304 https://registry.npmjs.org/part-file
npm http 304 https://registry.npmjs.org/speedometer
npm http 304 https://registry.npmjs.org/optimist
npm http 304 https://registry.npmjs.org/once
npm http 304 https://registry.npmjs.org/commander/0.6.1
npm http 304 https://registry.npmjs.org/connect/2.6.0
npm http 304 https://registry.npmjs.org/cookie/0.0.4
npm http 304 https://registry.npmjs.org/mkdirp/0.3.3
npm http 304 https://registry.npmjs.org/crc/0.2.0
npm http GET https://registry.npmjs.org/bncode
npm http GET https://registry.npmjs.org/compact2string
npm http GET https://registry.npmjs.org/thunky
npm http GET https://registry.npmjs.org/request
npm http GET https://registry.npmjs.org/wordwrap
npm http 304 https://registry.npmjs.org/fresh/0.1.0
npm http 304 https://registry.npmjs.org/methods/0.0.1
npm WARN package.json methods@0.0.1 No README.md file found!
npm http 304 https://registry.npmjs.org/debug
npm http 304 https://registry.npmjs.org/send/0.1.0
npm http 304 https://registry.npmjs.org/socket.io-client/0.9.16
npm http 304 https://registry.npmjs.org/base64id/0.1.0
npm http 304 https://registry.npmjs.org/policyfile/0.0.4
npm http 304 https://registry.npmjs.org/qs
npm http 304 https://registry.npmjs.org/redis/0.7.3
npm http GET https://registry.npmjs.org/mime/1.2.6
npm http 304 https://registry.npmjs.org/tunnel-agent
npm http 304 https://registry.npmjs.org/forever-agent
npm http 304 https://registry.npmjs.org/json-stringify-safe
npm http 304 https://registry.npmjs.org/aws-sign
npm http 304 https://registry.npmjs.org/http-signature
npm http 304 https://registry.npmjs.org/hawk
npm http 304 https://registry.npmjs.org/node-uuid
npm http 304 https://registry.npmjs.org/cookie-jar
npm http 304 https://registry.npmjs.org/oauth-sign
npm http 304 https://registry.npmjs.org/form-data/0.0.8
npm http GET https://registry.npmjs.org/qs/0.5.1
npm http GET https://registry.npmjs.org/formidable/1.0.11
npm http GET https://registry.npmjs.org/bytes/0.1.0
npm http GET https://registry.npmjs.org/send/0.0.4
npm http GET https://registry.npmjs.org/pause/0.0.1
npm http 304 https://registry.npmjs.org/less/1.3.3
npm http 304 https://registry.npmjs.org/cli-color/0.2.2
npm http 304 https://registry.npmjs.org/commander/1.1.1
npm http 304 https://registry.npmjs.org/wrench/1.5.1
npm http 304 https://registry.npmjs.org/requirejs/2.1.6
npm http GET https://registry.npmjs.org/assert-plus/0.1.2
npm http GET https://registry.npmjs.org/asn1/0.1.11
npm http GET https://registry.npmjs.org/ctype/0.5.2
npm http GET https://registry.npmjs.org/boom
npm http GET https://registry.npmjs.org/hoek
npm http GET https://registry.npmjs.org/cryptiles
npm http GET https://registry.npmjs.org/sntp
npm http GET https://registry.npmjs.org/combined-stream
npm http 304 https://registry.npmjs.org/bncode
npm http 304 https://registry.npmjs.org/thunky
npm http 304 https://registry.npmjs.org/request
npm http 304 https://registry.npmjs.org/compact2string
npm http 304 https://registry.npmjs.org/watchr/2.4.3
npm http 304 https://registry.npmjs.org/bytes/0.1.0
npm http 304 https://registry.npmjs.org/formidable/1.0.11
npm http 304 https://registry.npmjs.org/qs/0.5.1
npm http 304 https://registry.npmjs.org/mime/1.2.6
npm http 304 https://registry.npmjs.org/wordwrap
npm http GET https://registry.npmjs.org/keypress
npm http GET https://registry.npmjs.org/typechecker
npm http GET https://registry.npmjs.org/bal-util
npm http GET https://registry.npmjs.org/extendr
npm http GET https://registry.npmjs.org/eachr
npm http GET https://registry.npmjs.org/safefs
npm http GET https://registry.npmjs.org/taskgroup
npm http 304 https://registry.npmjs.org/assert-plus/0.1.2
npm http 304 https://registry.npmjs.org/pause/0.0.1
npm http 304 https://registry.npmjs.org/send/0.0.4
npm http 304 https://registry.npmjs.org/ctype/0.5.2
npm http 304 https://registry.npmjs.org/asn1/0.1.11
npm http GET https://registry.npmjs.org/es5-ext
npm http GET https://registry.npmjs.org/memoizee
npm http 304 https://registry.npmjs.org/combined-stream
npm http 304 https://registry.npmjs.org/sntp
npm http 304 https://registry.npmjs.org/boom
npm http 304 https://registry.npmjs.org/hoek
npm http 304 https://registry.npmjs.org/cryptiles
npm http GET https://registry.npmjs.org/form-data
npm WARN package.json cookie-jar@0.2.0 No README.md file found!
npm WARN package.json aws-sign@0.2.0 No README.md file found!
npm WARN package.json oauth-sign@0.2.0 No README.md file found!
npm WARN package.json forever-agent@0.2.0 No README.md file found!
npm WARN package.json tunnel-agent@0.2.0 No README.md file found!
npm http 304 https://registry.npmjs.org/eachr
npm http 304 https://registry.npmjs.org/extendr
npm http 304 https://registry.npmjs.org/bal-util
npm http 304 https://registry.npmjs.org/typechecker
npm http 304 https://registry.npmjs.org/keypress
npm http GET https://registry.npmjs.org/delayed-stream/0.0.5
npm http GET https://registry.npmjs.org/uglify-js/1.2.5
npm http GET https://registry.npmjs.org/ws
npm http GET https://registry.npmjs.org/xmlhttprequest/1.4.2
npm http GET https://registry.npmjs.org/active-x-obfuscator/0.0.1
npm http 304 https://registry.npmjs.org/safefs
npm http 304 https://registry.npmjs.org/form-data
npm http 304 https://registry.npmjs.org/taskgroup
npm http 304 https://registry.npmjs.org/memoizee
npm http 304 https://registry.npmjs.org/es5-ext
npm http 304 https://registry.npmjs.org/active-x-obfuscator/0.0.1
npm http 304 https://registry.npmjs.org/xmlhttprequest/1.4.2
npm http 304 https://registry.npmjs.org/uglify-js/1.2.5
npm http 304 https://registry.npmjs.org/ws
npm http 304 https://registry.npmjs.org/delayed-stream/0.0.5
npm http GET https://registry.npmjs.org/ambi
npm http GET https://registry.npmjs.org/eventemitter2
npm http GET https://registry.npmjs.org/zeparser/0.0.5
npm http GET https://registry.npmjs.org/event-emitter
npm http GET https://registry.npmjs.org/ycssmin
npm http GET https://registry.npmjs.org/next-tick
npm http GET https://registry.npmjs.org/tinycolor
npm http GET https://registry.npmjs.org/commander
npm http GET https://registry.npmjs.org/options
npm http 304 https://registry.npmjs.org/ambi
npm http 304 https://registry.npmjs.org/eventemitter2
npm http 304 https://registry.npmjs.org/tinycolor
npm http 304 https://registry.npmjs.org/next-tick
npm http 304 https://registry.npmjs.org/zeparser/0.0.5
npm http 304 https://registry.npmjs.org/event-emitter
npm http 304 https://registry.npmjs.org/ycssmin
npm http 304 https://registry.npmjs.org/options
npm http 304 https://registry.npmjs.org/commander

> ws@0.4.27 install /Users/rodrigoayala/Documents/workspace/Proyectos Personales/tv.js/node_modules/socket.io/node_modules/socket.io-client/node_modules/ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)

  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/bufferutil.node
  SOLINK_MODULE(target) Release/bufferutil.node: Finished
  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/validation.node
  SOLINK_MODULE(target) Release/validation.node: Finished
q@0.9.6 node_modules/q

underscore@1.4.4 node_modules/underscore

readable-stream@1.0.2 node_modules/readable-stream

fluent-ffmpeg@1.5.2 node_modules/fluent-ffmpeg
└── async@0.2.9

express@3.0.0 node_modules/express
├── methods@0.0.1
├── fresh@0.1.0
├── range-parser@0.0.4
├── cookie@0.0.4
├── crc@0.2.0
├── commander@0.6.1
├── debug@0.7.2
├── mkdirp@0.3.3
├── send@0.1.0 (mime@1.2.6)
└── connect@2.6.0 (pause@0.0.1, bytes@0.1.0, formidable@1.0.11, qs@0.5.1, send@0.0.4)

request@2.22.0 node_modules/request
├── json-stringify-safe@4.0.0
├── aws-sign@0.3.0
├── forever-agent@0.5.0
├── qs@0.6.5
├── tunnel-agent@0.3.0
├── oauth-sign@0.3.0
├── cookie-jar@0.3.0
├── node-uuid@1.4.0
├── mime@1.2.9
├── http-signature@0.10.0 (assert-plus@0.1.2, asn1@0.1.11, ctype@0.5.2)
├── hawk@0.13.1 (cryptiles@0.2.2, hoek@0.8.5, sntp@0.2.4, boom@0.4.2)
└── form-data@0.0.8 (async@0.2.9, combined-stream@0.0.4)

peerflix@0.1.1 node_modules/peerflix
├── bitfield@0.1.0
├── speedometer@0.1.1
├── clivas@0.1.4
├── once@1.1.1
├── range-parser@0.0.4
├── peer-wire-protocol@0.3.8
├── hat@0.0.3
├── network-address@0.0.4
├── mime@1.2.9
├── numeral@1.4.8
├── part-file@0.1.0 (thunky@0.1.0)
├── optimist@0.3.7 (wordwrap@0.0.2)
├── peer-wire-swarm@0.0.8 (compact2string@1.0.0, bncode@0.2.3)
└── read-torrent@0.1.0 (bncode@0.2.3, request@2.16.6)

yapp@0.0.1 node_modules/yapp
├── wrench@1.5.1
├── commander@1.1.1 (keypress@0.1.0)
├── requirejs@2.1.6
├── watchr@2.4.3 (typechecker@2.0.1, eachr@2.0.2, safefs@3.0.1, extendr@2.0.1, bal-util@2.0.5, taskgroup@3.1.2)
├── cli-color@0.2.2 (es5-ext@0.9.2, memoizee@0.2.4)
└── less@1.3.3 (ycssmin@1.0.1)

socket.io@0.9.16 node_modules/socket.io
├── base64id@0.1.0
├── policyfile@0.0.4
├── redis@0.7.3
└── socket.io-client@0.9.16 (xmlhttprequest@1.4.2, uglify-js@1.2.5, active-x-obfuscator@0.0.1, ws@0.4.27)
```

```
$ make
```

**Output of the previous command:**

```
node_modules/yapp/bin/yapp.js -d public/main build
[load] Loading /Users/rodrigoayala/Documents/workspace/Proyectos Personales/tv.js/public/main/build.js ...
[build] Start building application Main
[build] Build directory : /Users/rodrigoayala/Documents/workspace/Proyectos Personales/tv.js/public/main/build
[build]
[html] Generating code for application
[html] End of generation
[html]
[js] Generating code for application
[js]
[stylesheet] Generating style for application
[stylesheet]
[static] Generating static files for application
[static]   -> Creating templates
[static]   -> Creating images
[static] End of generation
[static]
node_modules/yapp/bin/yapp.js -d public/remote build
[load] Loading /Users/rodrigoayala/Documents/workspace/Proyectos Personales/tv.js/public/remote/build.js ...
[build] Start building application Remote
[build] Build directory : /Users/rodrigoayala/Documents/workspace/Proyectos Personales/tv.js/public/remote/build
[build]
[html] Generating code for application
[html] End of generation
[html]
[js] Generating code for application
[js]
[stylesheet] Generating style for application
[stylesheet]
[static] Generating static files for application
[static]   -> Creating templates
[static] End of generation
[static]
/usr/local/bin/node bin/run.js
Server listening on: http://localhost:8888

```

**Tree of files and folders:**
https://gist.github.com/RodrigoAyala/5976846

**Comment by mojocatplay** [comment]
```
[js] Generating code for application
[js]
[stylesheet] Generating style for application
[stylesheet]
```

I believe you could encounter the same problem with me: the absolute path to the files (in your case, and mine, the path to build application.js and application.css file) to build contains space character and it breaks the command.

Took me half an hour to find out the problem, since the yapp build did not output the error message correctly (in this case log.js(err.stdout, "error") output nothing, so I changed it to log.js(err, "error") in order for it to show the error)

**Comment by RodrigoAyala** [comment]
Hi!, thank you for helping me to try to fix this issue!.

With your help I've realized that it's a yapp.js problem related to spaced paths.  They're using the exec method from child_process module, which is a binding to make system calls, and this uses /bin/sh. And sh don't know how to handle commands located on paths with spaces!.  Therefore, I located the command (actually yapp,js calls two compilers), and I've surrounded the compilers location with quotes, and it works quite well.  I made a pull request with that: (https://github.com/FriendCode/yapp.js/pull/4).

This issue will be solved when that pull request is accepted.

Should I close this issue or I must do it when the problem is solved on the yapp.js project?

Cheers,
Rodrigo
